### PR TITLE
[LAB-4] Migrations 

### PR DESCRIPTION
**Pull request by JJerome-NM**

### CODE REVIEW

File: src/main/resources/db/changelog/1_setup.xml [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        </createTable>
+        
+        <createTable tableName="course_member">
+            <column name="user_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="course_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="member_type" type="varchar(25)">
```
Comment (Marko-Shevchuk) [inline src/main/resources/db/changelog/1_setup.xml]: 20

**Review by Marko-Shevchuk — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/1_setup.xml [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            <column name="user_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="parent_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="turned_in_date" type="datetime"/>
+            <column name="grade" type="int"/>
+            <column name="auto_grade" type="int"/>
+            <column name="create_date" type="datetime">
```
Comment (ArsenOOO7) [inline src/main/resources/db/changelog/1_setup.xml:134]: created_date

File: src/main/resources/db/changelog/1_setup.xml [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                <constraints nullable="false"/>
+            </column>
+            <column name="update_date" type="datetime">
```
Comment (ArsenOOO7) [inline src/main/resources/db/changelog/1_setup.xml:137]: updated_date

File: src/main/resources/db/changelog/1_setup.xml [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            <column name="assignment_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="user_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="parent_id" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="turned_in_date" type="datetime"/>
```
Comment (Marko-Shevchuk) [inline src/main/resources/db/changelog/1_setup.xml:131]: turned_in varchar(20) is missing

File: src/main/resources/db/changelog/1_setup.xml [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+                <constraints nullable="false"/>
+            </column>
+            <column name="message" type="varchar(1000)">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+        <createIndex tableName="submission_comment" indexName="submission_comment_idx1">
+            <column name="submission_id"/>
+        </createIndex>
+        <createIndex tableName="submission_comment" indexName="submission_comment_idx2">
```
Comment (ArsenOOO7) [inline src/main/resources/db/changelog/1_setup.xml:174]: remove

**Review by Marko-Shevchuk — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/1_setup.xml [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                <constraints nullable="false"/>
+            </column>
+            <column name="deleted" type="boolean" defaultValue="false">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+        <createIndex tableName="assignment_comment" indexName="assignment_comment_idx1">
+            <column name="assignment_id"/>
+        </createIndex>
+        <createIndex tableName="assignment_comment" indexName="assignment_comment_idx2">
```
Comment (ArsenOOO7) [inline src/main/resources/db/changelog/1_setup.xml:114]: remove

File: src/main/resources/db/changelog/1_setup.xml [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+                                 referencedColumnNames="id"/>
+
+<!--        submission              -->
+
+        <addForeignKeyConstraint baseTableName="submission"
+                                 baseColumnNames="assignment_id"
+                                 constraintName="submission_fk1"
+                                 referencedTableName="assignment"
+                                 referencedColumnNames="id"/>
+
```
Comment (ArsenOOO7) [inline src/main/resources/db/changelog/1_setup.xml:242]: Add submission fk (```parent_id``` to ```id``` in submission)

**Review by ArsenOOO7 — CHANGES_REQUESTED** [review]
Good Job!

**Review by ArsenOOO7 — APPROVED** [review]

File: src/main/resources/db/changelog/1_setup.xml [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+                                 constraintName="submission_attached_content_fk1"
+                                 referencedTableName="submission"
+                                 referencedColumnNames="id"/>
+
+        <addForeignKeyConstraint baseTableName="assignment_attached_content"
+                                 baseColumnNames="assignment_id"
+                                 constraintName="assignment_attached_content_fk1"
+                                 referencedTableName="assignment"
+                                 referencedColumnNames="id"/>
+
```
Comment (Marko-Shevchuk) [inline src/main/resources/db/changelog/1_setup.xml:267]: course_member FK course_id is missing

**Review by Marko-Shevchuk — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/1_setup.xml [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+        </createTable>
+
+        <createIndex tableName="submission_attached_content" indexName="submission_attached_content_idx1">
+            <column name="submission_id"/>
+        </createIndex>
+    </changeSet>
+
+    <changeSet id="add_setup_fk" author="JJerome">
+
+<!--        assignment               -->
```
Comment (ArsenOOO7) [inline src/main/resources/db/changelog/1_setup.xml:210]: fix comment

**Review by Marko-Shevchuk — APPROVED** [review]

**Pull request merged**